### PR TITLE
Port a number of task interfaces from Nipype

### PR DESCRIPTION
**Pull request by Mahdieh-dst**
Created a Pydra Task interface in a module named Preprocess.py for ANTSRegistration command.

### CODE REVIEW

**Comment by Mahdieh-dst** [comment]
**pydra/tasks/ants/utils:**  successfully ported and tested ai.py, imagemath.py, multiplyimages.py, thresholdimage.py
**pydra/tasks/ants/segmentation:** successfully ported and tested atropos.py, n4biasfieldcorrection.py
**pydra/tasks/ants/registeration:** successfully ported and tested registeration_tested_preprocess

**Comment by tclose** [comment]
@Mahdieh-dst the tests are failing. Could you please add a file called `pyproject.toml` to the root directory with the following contents to your branch?

```
[build-system]
requires = ["setuptools==62", "wheel"]
```

File: pydra/tasks/ants/registration/antsRegisteration_help.txt [hunk 1]
```diff
@@ -0,0 +1,236 @@
+➜  ~ export ANTSPATH=/Users/mahdieh/install/bin/
```
Comment (tclose) [inline pydra/tasks/ants/registration/antsRegisteration_help.txt:1]: Could you remove this file from your branch before I merge it?

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (tclose) [inline .vscode/launch.json:1]: could you remove the .vscode directory with `rm -r --cache .vscode` and then add `/.vscode` to the `.gitignore` file?

File: pydra/tasks/ants/registration/registration_ported.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
+from pydra.engine import specs
```
Comment (tclose) [inline pydra/tasks/ants/registration/registration_ported.py:1]: Should this file be renamed to just `registration.py`?

**Review by tclose — CHANGES_REQUESTED** [review]
Looks great, are there tests that actually run the commands as well (great work if so)?

File: pydra/tasks/ants/registration/registration_ported.py (hunk 3 above)
Comment (Mahdieh-dst) [reply pydra/tasks/ants/registration/registration_ported.py:1]: No, "registeration_ported.py" should be deleted and "registeration_tested_preprocess.py" should be renamed to registration.py. I will do make the changes

File: .vscode/launch.json (hunk 2 above)
Comment (Mahdieh-dst) [reply .vscode/launch.json:1]: Yep, will make the changes

File: pydra/tasks/ants/registration/antsRegisteration_help.txt (hunk 1 above)
Comment (Mahdieh-dst) [reply pydra/tasks/ants/registration/antsRegisteration_help.txt:1]: Sure, will do

**Review by tclose — APPROVED** [review]

**Comment by tclose** [comment]
> What's the state of pydra for doing nipype workflows? I see most of the packages like this are stubs. I'd like to implement a pipeline using the new pydra-ants interfaces.

hi @gdevenyi, we are actively working on this and hope to have more than a few stubs available soon. There is a prototype nipype-to-pydra converter you could use if you wanted to try it out to convert the interfaces that you need (https://github.com/nipype/nipype2pydra).

Have a look at https://github.com/nipype/nipype2pydra/tree/main/example-specs/task for some example specs to use as a template.

**Pull request closed without merging**